tests/ssl-opt.sh

```diff
@@ -691,19 +701,19 @@ run_test() {
 
         # run the commands
         if [ -n "$PXY_CMD" ]; then
-            echo "$PXY_CMD" > $PXY_OUT
+            printf "# $NAME\n$PXY_CMD\n" > $PXY_OUT
```

Comment:
This is wrong, and likewise for the other changes to call `printf`. Calling `echo` is fine here since the argument doesn't start with `-` and, in practice, doesn't contain backslashes. Calling `printf` breaks because the argument sometimes contains `%s`.

```
Renegotiation: gnutls server strict, client-initiated .................. tests/ssl-opt.sh: 711: printf: %S: invalid directive
PASS
Renegotiation: gnutls server unsafe, client-initiated default .......... tests/ssl-opt.sh: 711: printf: %D: invalid directive
PASS
Renegotiation: gnutls server unsafe, client-inititated no legacy ....... tests/ssl-opt.sh: 711: printf: %D: invalid directive
PASS
Renegotiation: gnutls server unsafe, client-inititated legacy .......... tests/ssl-opt.sh: 711: printf: %D: invalid directive
PASS
```

Reply:
I was using `printf` because IIRC the interpretation of `\n` in the argument of `echo` is unspecified, but indeed the way I've been using it is completely wrong. I'll fix that.

Reply:
Right, to embed a newline with `echo` you either need separate `echo` commands or a literal newline. `echo '\n'` either prints a newline or `\n` depending on the shell.

Reply:
Fixed, is now ready to review again.